server/src/shared/interfaces/response-objects/BoardResponseObject.ts

```diff
@@ -12,6 +12,8 @@ export interface BoardResponseObject {
     moderator: string[];
   };
   slides: string[] | Types.ObjectId[];
+  currentUserPermission: string,
```

Comment:
Should we use a string here? Since we have specific permissions, it’s not just any text. Perhaps it would be better to use an enum, something like this:

```typescript
export enum BoardPermission {
  NONE = "NONE",
  VIEWER = "VIEWER",
  EDITOR = "EDITOR",
  MODERATOR = "MODERATOR",
  OWNER = "OWNER",
}
```

Reply:
It's not possible to change it to str-enums with current design in auth and permissions validation 
![image](https://github.com/user-attachments/assets/292131ea-4f51-4f18-8075-e2af177c53a4)